include/dkm.hpp

```diff
@@ -183,7 +183,7 @@ used for initializing the means.
 */
 template <typename T, size_t N>
 std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>> kmeans_lloyd(
-	const std::vector<std::array<T, N>>& data, uint32_t k) {
+	const std::vector<std::array<T, N>>& data, uint32_t k, int maxIter) {
```

Comment:
1. Please add an additional argument:

`float epsilon = 0.0f`

2. Please add a new method after the `T distance_squared` declaration:

```
    /*
     Calculate the mean square float distance between a collection of points.
     */
    template <typename T, size_t N>
    float point_collection_epsilon(const std::vector< std::array<T, N> >& point_a, const std::vector< std::array<T, N> >& point_b) {
        assert( point_a.size() == point_b.size() );
        float d_squared = 0.0f;
        for( typename std::vector< std::array<T, N> >::size_type pointNum = 0; pointNum < point_a.size(); ++pointNum) {
            
            float point_d = distance_squared( point_a[pointNum], point_b[pointNum] );
            d_squared += point_d/N;
        }
        d_squared /= point_a.size();
        return d_squared;
    }
```

3. Change the `means != old_means` in line 202 into 
`details::point_collection_epsilon(means, old_means > epsilon`

Although it might be a data-centric parameter, hard to evaluate, the default zero value is basically a better way to handle the `means != old_means` check.